### ISSUE TITLE
[Bug]: New Listener Client API not properly receiving JSON object from client... I think...

### ISSUE THREAD

**Issue by mattv8** [josephdadams/TallyArbiter#305]
### What happened?

In updating my listener client to work with the new Listener Client API, I'm having trouble getting TA to receive my JSON object. I am able to establish the connection just fine, but TA doesn't pull in any of the JSON variables and uses defaults.

From my C# client, the SocketIO emit statement is pretty straightforward (paraphrasing my code for clarity):
```
foreach (Devices device in deviceList) {
    socket.Emit("listenerclient_connect", deviceObj);
}
```
Where `deviceObj` are well-formed JSON strings, and look like the following (using a deviceList containing "Cam 1" and "Cam 2" deviceId's as an example):
```
{"deviceId":"e8a2bd0b","listenerType":"OpenTally Cam 1","canBeReassigned":true,"canBeFlashed":true,"supportsChat":false}
{"deviceId":"9ff13e3d","listenerType":"OpenTally Cam 2","canBeReassigned":true,"canBeFlashed":true,"supportsChat":false}
```

However, TA establishes the connections with default listenerclient_connect variables:
![image](https://user-images.githubusercontent.com/9312603/146434938-1783433c-fd5a-46be-b924-4c2065924e71.png)

I have also tried slightly different client syntax like the following with the same issue:
```
foreach (Devices device in deviceList) {
    client.listenerType = "OpenTally " + device.name;
    client.canBeReassigned = true;
    client.canBeFlashed = true;
    client.supportsChat = false;
    socket.Emit("listenerclient_connect", device.id, listenerType, canBeReassigned, canBeFlashed, supportsChat);
}
```

Any tips to help me debug?

### Version

3.0.1

### Distribution

Desktop

### OS

Windows

### TallyArbiter configuration

```javascript
{
  "security": {
    "username_producer": "producer",
    "password_producer": "12345",
    "username_settings": "admin",
    "password_settings": "12345"
  },
  "cloud_destinations": [],
  "cloud_keys": [],
  "device_actions": [],
  "device_sources": [],
  "devices": [
    {
      "name": "Cam 1",
      "description": "Cam 1 Tally",
      "linkedBusses": [
        "e393251c",
        "334e4eda"
      ],
      "enabled": true,
      "id": "e8a2bd0b"
    },
    {
      "name": "Cam 2",
      "description": "Cam 2 Tally",
      "linkedBusses": [
        "e393251c",
        "334e4eda"
      ],
      "enabled": true,
      "id": "9ff13e3d"
    }
  ],
  "sources": [],
  "tsl_clients": [],
  "tsl_clients_1secupdate": false,
  "bus_options": [
    {
      "id": "e393251c",
      "label": "Preview",
      "type": "preview",
      "color": "#3fe481",
      "priority": 50
    },
    {
      "id": "334e4eda",
      "label": "Program",
      "type": "program",
      "color": "#e43f5a",
      "priority": 200
    },
    {
      "id": "12c8d699",
      "label": "Aux 1",
      "type": "aux",
      "color": "#0000FF",
      "priority": 100
    },
    {
      "id": "12c8d689",
      "label": "Aux 2",
      "type": "aux",
      "color": "#0000FF",
      "priority": 100
    }
  ],
  "externalAddress": "http://0.0.0.0:4455/#/tally",
  "remoteErrorReporting": false,
  "uuid": "1a2d8c40"
}
```

### Relevant log output

```shell
[12/16/21 12:19 PM] Listener Client Connected. Type: client Device: Cam 1
[12/16/21 12:19 PM] Sending device states to Listener Client: 4cc5c237 - Cam 1
[12/16/21 12:19 PM] Listener Client Connected. Type: client Device: Cam 1
[12/16/21 12:19 PM] Sending device states to Listener Client: 4cc5c237 - Cam 1
[12/16/21 12:19 PM] Sending device states to Listener Client: 9258a705 - Cam 1
```

**Comment by josephdadams** [comment]
Wondering if you could run TA from source and console log what is being received. Maybe it’s not being parsed as an object for some reason.

**Comment by mattv8** [comment]
Okay @josephdadams, I have had the most busy Dec - Feb with work I think I've ever had. Things are finally slowing down again so I'm getting back into this.

I had a chance to rebuild TA from source in a dev environment and add an extra line for debugging. TA is indeed receiving my JSON objects as well-formed, but it's still only recognizing it as the same device in TA's UI. Here's the output of TA's log upon connect:
```
[2022-03-02 11:16:53] info: Received JSON object: {"deviceId":"7714064f","listenerType":"OpenTally_Cam_1","canBeReassigned":true,"canBeFlashed":true,"supportsChat":false}
[2022-03-02 11:16:53] info: Listener Client Connected. Type: client Device: Cam 1
[2022-03-02 11:16:53] info: Sending device states to Listener Client: 1337566d - Cam 1
[2022-03-02 11:16:53] info: Received JSON object: {"deviceId":"3bcc67eb","listenerType":"OpenTally_Cam_2","canBeReassigned":true,"canBeFlashed":true,"supportsChat":false}
[2022-03-02 11:16:53] info: Listener Client Connected. Type: client Device: Cam 1
[2022-03-02 11:16:53] info: Sending device states to Listener Client: 1337566d - Cam 1
```
Then in TA's UI still only a duplicate of Cam1 is seen, and it's not reassignable despite canBeReassigned=true:
![image](https://user-images.githubusercontent.com/9312603/156424421-c313dd46-9704-4a31-87a8-3dad98ad9653.png)

**Comment by mattv8** [comment]
Okay I think I have been able to get it working. The issue is indeed with TA, for some reason the JSON isn't being parsed. I added a line of code within the `socket.on('listenerclient_connect', function(obj: ListenerClientConnect) { }` function to basically re-parse the JSON as it doesn't seem to be parsed the first go around. Would love to have someone take a look and see if there is a more elegant solution than what I've implemented.

PR #393
